core/include/detray/utils/tuple.hpp

```diff
@@ -40,10 +46,59 @@ struct tuple<T, Ts...> {
         : v(std::forward<U>(_v)),
     r(std::forward<Us>(_r)...) {}
 
+    DETRAY_HOST_DEVICE constexpr tuple(const tuple &o) requires(
+        std::is_copy_constructible_v<T> &&
+        (std::is_copy_constructible_v<Ts> && ...))
+        : v(o.v), r(o.r) {}
+
+    DETRAY_HOST_DEVICE constexpr tuple(tuple &&o) requires(
+        std::is_move_constructible_v<T> &&
+        (std::is_move_constructible_v<Ts> && ...))
+        : v(std::move(o.v)), r(std::move(o.r)) {}
```

Comment:
😕 Do the templated copy and move constructors not cover this use case?

Though I remember that under some circumstances compilers indeed seem to have issues with this... 🤔

Reply:
Honestly, probably? But the standard library explicitly defines both of these for tuple, so I thought I'd do the same.

Reply:
Ah yes, SYCL needs these types to be trivially copyable, so we need to allow defaulted copy operators (which differ from the templated ones).